addons/website_sale_stock/models/sale_order.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _cart_update(self, product_id=None, line_id=None, add_qty=0, set_qty=0, **kwargs):
+        values = super(SaleOrder, self)._cart_update(product_id, line_id, add_qty, set_qty, **kwargs)
+        line_id = values.get('line_id')
+        for line in self.order_line:
```

Comment:
still think we do some unnecessary verifications here but it shouldn't be a big deal, carts shouldn't have a huge amount of lines.